### PR TITLE
QF-3716 Add possibility to rerun job in the Admin

### PR DESCRIPTION
**Pull request by dddpt**
The re-run button sets the job back to "pending" status.

However, on my local setup, jobs are systematically failing as the projects are missing their files.
Requires maybe some more tests?

Screenshot from 2024-02-14 17-32-26

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-3716 Add possibility to rerun job in the Admin](https://app.clickup.com/t/2192114/QF-3716)

File: docker-app/qfieldcloud/core/admin.py [hunk 1]
```diff
@@ -869,6 +876,10 @@ def export_applyjob_deltafile(
             },
         )
 
+    def rerun_job(self, request, object_id):
+        Job.objects.filter(pk=object_id).update(status="pending")
+        return redirect(request.META.get("HTTP_REFERER", "../.."))
```
Comment (suricactus) [inline docker-app/qfieldcloud/core/admin.py:881]: I guess the intention here is to return to the root of the admin? Does it really make sense? Why don't you stay on current job page, so you can see how it progress in statuses?

Also, I believe there is a better mechanism to move to another page than using the headers. Check the web UI views for examples.

File: docker-app/qfieldcloud/core/templates/admin/job_change_form.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{% extends 'admin/change_form.html' %}
+
+{% block submit_buttons_bottom %}
+  {{ block.super }}
+
+  <div class="submit-row">
+    {% if original %}
+        <a href="{% url 'admin:rerun_job' original.pk %}" class="button">Re-run</a>
```
Comment (suricactus) [inline docker-app/qfieldcloud/core/templates/admin/job_change_form.html:8]:
```suggestion
        <a href="{% url 'admin:rerun_job' original.pk %}" class="button">{% trans 'Re-run job' %}</a>
```

NOTE this needs the`i18n` to be imported in this template.

File: docker-app/qfieldcloud/core/admin.py (hunk 1 above)
Comment (dddpt) [reply docker-app/qfieldcloud/core/admin.py:881]: Oops, that was a leftover from a different approach I tried, `return redirect("..")` is the simplest and sufficient.

Comment (suricactus) [reply docker-app/qfieldcloud/core/admin.py:881]: `HttpResponseRedirect(".")` is something that was already used as a pattern in the code

File: docker-app/qfieldcloud/core/templates/admin/job_change_form.html (hunk 2 above)
Comment (dddpt) [reply docker-app/qfieldcloud/core/templates/admin/job_change_form.html]: done

**Review by suricactus — APPROVED** [review]

**Pull request merged**